docs/create_your_own_brick.rst

```diff
@@ -0,0 +1,371 @@
+later, after the creation of the brick. To enable this mechanism,
+called lazy initialization, you need to decorate the constructor with the 
+:func:`.lazy` decorator:
+
+    >>> @lazy(allocation=['attr1', 'attr2']) # doctest: +SKIP
+    ... def __init__(self, attr1, attr1)
+    ...     ...
+
+This allows the user to specify ``attr1`` and ``attr2`` after the creation of 
+the brick. For example, the following ``DoubleSequential`` brick is composed of
+two :class:`.Feedforward` bricks for which you do not need to
+specify the ``input_dim`` of ``brick2`` directly at its creation.
+
+    >>> class DoubleSequential(Feedforward):
+    ...     """Two sequential Feedforward bricks."""
+    ...     def __init__(self, brick1, brick2, **kwargs):
+    ...         super(Feedforward, self).__init__(**kwargs)
+    ...         self.brick1 = brick1
+    ...         self.brick2 = brick2
+    ...         self.input_dim = brick1.input_dim
+    ...         self.output_dim = brick2.output_dim
+    ...         self.children = [self.brick1, self.brick2]
+    ...
+    ...     def _push_allocation_config(self):
+    ...         self.brick2.input_dim = self.brick1.get_dim('output')
+    ...
+    ...     @application
+    ...     def apply(self, x):
+    ...         return self.brick2.apply(self.brick1.apply(x))
+
+Note how ``get_dim`` is used to retrieve the ``input_dim`` of ``brick1``. You
+can now use a ``DoubleSeuential`` brick as follows.
+
+    >>> brick1 = Linear(input_dim=3, output_dim=2, use_bias=False,
+    ...                 weights_init=Constant(2))
+    >>> brick2 = Linear(output_dim=4, use_bias=False, weights_init=Constant(2))
+    >>>
+    >>> seq = DoubleSequential(brick1, brick2)
+    >>> seq.initialize()
+    >>> brick2.input_dim
+    2
+
+
+Example
+-------
+
+.. doctest::
+   :hide:
+
+    >>> import numpy
+    >>>
+    >>> import theano
+    >>> from theano import tensor
+    >>>
+    >>> from blocks.bricks import Initializable, Linear
+    >>> from blocks.bricks.base import lazy, application
+    >>> from blocks.bricks.parallel import Parallel
+    >>> from blocks.initialization import Constant
+    >>> from blocks.roles import add_role, WEIGHT
+    >>> from blocks.utils import shared_floatx_nans
+
+For the sake of the tutorial, let's consider a toy operation that takes two
+batch inputs and multiply them respectively by two matrices, resulting in two
+outputs.
+
+The first step is to identify which brick to inherit from. Clearly we are
+implementing a variant of the :class:`.Linear` brick. Contrary to
+:class:`.Linear`, ours has two inputs and two outputs, which means that we can
+not inherit from :class:`.Feedforward`, which requires a single input and a
+single output. Our brick will have to manage two shared variables
+representing the matrices to multiply the inputs with and thus, inheriting from
+:class:`.Initializable` makes perfectly sense as we will let the user decide
```

Comment:
Inheriting from `Initializable` just adds `weights_init` and `biases_init` in this case, I am not sure I understand what "let the user decide which initialization scheme to use" means.